Stripe/STPSourceOwner.m

```diff
@@ -29,26 +29,28 @@ @implementation STPSourceOwner
 
 #pragma mark - STPAPIResponseDecodable
 
-+ (NSArray *)requiredFields {
-    return @[];
-}
-
 + (instancetype)decodedObjectFromAPIResponse:(NSDictionary *)response {
-    NSDictionary *dict = [response stp_dictionaryByRemovingNullsValidatingRequiredFields:[self requiredFields]];
+    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
     if (!dict) {
         return nil;
     }
 
     STPSourceOwner *owner = [self new];
     owner.allResponseFields = dict;
-    owner.address = [STPAddress decodedObjectFromAPIResponse:dict[@"address"]];
-    owner.email = dict[@"email"];
-    owner.name = dict[@"name"];
-    owner.phone = dict[@"phone"];
-    owner.verifiedAddress = [STPAddress decodedObjectFromAPIResponse:dict[@"verified_address"]];
-    owner.verifiedEmail = dict[@"verified_email"];
-    owner.verifiedName = dict[@"verified_name"];
-    owner.verifiedPhone = dict[@"verified_phone"];
+    NSDictionary *rawAddress = [dict stp_dictionaryForKey:@"address"];
+    if (rawAddress) {
+        owner.address = [STPAddress decodedObjectFromAPIResponse:rawAddress];
+    }
+    owner.email = [dict stp_stringForKey:@"email"];
+    owner.name = [dict stp_stringForKey:@"name"];
+    owner.phone = [dict stp_stringForKey:@"phone"];
+    NSDictionary *rawVerifiedAddress = [dict stp_dictionaryForKey:@"verified_address"];
+    if (rawVerifiedAddress) {
```

Comment:
I believe all of the `decodedObjectFromAPIResponse:` methods gracefully handle `nil`.

Conversely, you don't check for non-`nil` strings before calling `fundingFromString:`, `brandFromString:`, etc. Those methods (so far) *also* gracefully handle `nil`.

I think this is totally fine as-is. I think I would slightly prefer *not* checking the dict against nil, because we have precedent that the decoding methods accept nil and it slightly reduces the amount of code.

I probably wouldn't change this (effort vs reward), but I will mention it in case there's something I'm missing or in case you feel more strongly about it.

Reply:
ah i think you're right. I can remove these checks